connection.go

```diff
@@ -26,6 +30,14 @@ const (
 	sessionClientSessionKeepAlive = "client_session_keep_alive"
 )
 
+var (
+	// FetchQueryMonitoringDataThresholdMs specifies the ms threshold, over which we'll fetch the monitoring
+	// data for a Snowflake query. We use a time-based threshold, since there is a non-zero latency cost
+	// to fetch this data and we want to bound the additional latency. By default we bound to a 2% increase
+	// in latency - assuming worst case 100ms - when fetching this metadata.
+	FetchQueryMonitoringDataThresholdSec float64 = 5
```

Comment:
should this be a time.Duration ?

Reply:
Should this actually be a utils config value? (that way we can set it via env vars if needed).

Reply:
I'll change it to a Duration. 

And yes in our code, but no here. I do like to keep a separation between this lib and our internal code.